src/linter/root.go

```diff
@@ -921,6 +928,22 @@ func (d *RootWalker) parseFuncArgs(params []node.Node, parTypes phpDocParamsMap,
 	return args, minArgs
 }
 
+func (d *RootWalker) enterNamespace(ns *stmt.Namespace) bool {
+	name, ok := ns.NamespaceName.(*name.Name)
+	if !ok {
+		return true
+	}
+
+	nsName := meta.NameToString(name)
+	pathSuffix := namespaceKey(nsName)
+	path := namespaceKey(filepath.Dir(d.filename))
+	if !strings.HasSuffix(path, pathSuffix) {
+		d.Report(ns, LevelDoNotReject, "nameFilepath", "namespace %s name doesn't match file system path", nsName)
```

Comment:
Actually there is no rule in PSR-4 that would require full namespace to match the path in the filesystem. E.g. "./acme-log-writer/lib/File_Writer.php" is a valid path for class "\Acme\Log\Writer\File_Writer"

I would suggest that we either:

1. Make this configurable (I am afraid there can be too many options available though)
2. Move this to a custom check

You can read more about PSR-4 at https://www.php-fig.org/psr/psr-4/ .